services/sources/publisher_backlist_service.py

```diff
@@ -23,6 +24,48 @@ def __init__(self):
         
         self.airtable_auth_token = os.environ.get('AIRTABLE_KEY', None)
 
+    def delete_records(
+        self,
+        offset: Optional[int]=None,
+        limit: Optional[int]=None
+    ):
+        filter_by_formula = self.build_filter_by_formula_parameter(deleted=True, full_import=False, start_timestamp=None)
+        
+        array_json_records = self.get_records_array(limit, filter_by_formula)
+
+        for json_dict in array_json_records:
+            if json_dict['records']:
+                for records_value in json_dict['records']:
+                    try:
+                        record_metadata_dict = records_value['fields']
+                        pub_backlist_record = PublisherBacklistMapping(record_metadata_dict)
+                        pub_backlist_record.applyMapping()
+                        self.delete_manifest(record_metadata_dict, pub_backlist_record.record)
+                        self.delete_digital_asset_file()
+                    except Exception:
```

Comment:
Do we know for sure that the record has not been created if an exception has been thrown? could there be other exceptions that may happen?

Reply:
With the mapping creation being deleted, I think a better exception would be stating that 'Delete manifest method has failed for {records_value}'